templates/nvidia.yaml

```diff
@@ -54,4 +51,16 @@ spec:
         initContainers:
         - name: init-wait
           image: busybox
-          command: ['sh', '-c', 'sleep 90']
+          command: ['sh', '-c', 'sleep 180']
```

Comment:
Where do these sleeps come from? What are they waiting for?

It's generally preferable to make waits deterministic when possible.

Reply:
I understand. However, the device plugin depends on the driver starts the server (which is after it compiles the kernel module and set up everything needed during runtime). There is supposed to be some endpoints to indicate the readiness of the driver in runtime but I haven't had time to figure it out yet and the driver is not open sourced.

We don't necessarily to add the sleep here - the device plugin will just keep crash looping until the driver starts to be functioning. However, the sleep here would avoid that noise.

Reply:
Can we add a jira to come back at this. Waiting potentially multiples of 3 minutes to start a workload that depends on GPU would make me frustrated, as an operator.

Not a blocker for this PR.

Reply:
https://jira.mesosphere.com/browse/DCOS-62148

It is orthogonal to the GPU workloads. GPU workloads can be spun up immediately. This is just the device plugin to sleep for 180 seconds when the Nvidia addon is installed. Even without this sleep, the plugin has to wait for the driver to be ready, which usually take 90 ~ 180 seconds (compile kernel modules during container runtime). So, the device plugin somehow has to wait.